sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -673,55 +722,69 @@ def plot(self, ax=None, n_cols=3, line_kw=None, contour_kw=None):
                 preds = pd_result.individual
 
             if len(values) == 1:
-                if self.kind == 'individual' or self.kind == 'both':
-                    n_samples = self._get_sample_count(
-                        len(preds[self.target_idx])
-                    )
+                if self.kind in ("individual", "both"):
                     ice_lines = preds[self.target_idx]
-                    sampled = ice_lines[np.random.choice(
-                        ice_lines.shape[0], n_samples, replace=False
-                    ), :]
-                    for j, ins in enumerate(sampled):
-                        lines_ravel[i * j + j] = axi.plot(
-                            values[0], ins.ravel(), **individual_line_kw
+                    ice_lines_idx = rng.choice(
+                        ice_lines.shape[0], n_ice_to_plot, replace=False
+                    )
+                    ice_lines_subsampled = ice_lines[ice_lines_idx, :]
+                    for ice_idx, ice in enumerate(ice_lines_subsampled):
+                        line_idx = pd_plot_idx * n_lines + ice_idx
+                        lines_ravel[line_idx] = axi.plot(
+                            values[0], ice.ravel(), **individual_line_kw
                         )[0]
```

Comment:
For readability, could you please extract this block in a private helper method named `_plot_ice_lines` that would do both the sub-sampling and the actual plotting?

This would make it possible to reduce the indentation level and make formatting nicer as well.